modules/moordyn/src/MoorDyn.f90

```diff
@@ -1590,8 +1591,8 @@ SUBROUTINE MD_Init(InitInp, u, p, x, xd, z, other, y, m, DTcoupling, InitOut, Er
       IF (wordy > 0) print *, "allocating state vectors to size ", Nx
 
       ! allocate state vector and temporary state vectors based on size just calculated
-      ALLOCATE ( x%states(m%Nx), m%xTemp%states(m%Nx), m%xdTemp%states(m%Nx), STAT = ErrStat )
-      IF ( ErrStat /= ErrID_None ) THEN
+      ALLOCATE ( x%states(m%Nx), m%xTemp%states(m%Nx), m%xdTemp%states(m%Nx), STAT = ErrStat2 )
+      IF ( ErrStat2 /= ErrID_None ) THEN
         ErrMsg  = ' Error allocating state vectors.'
```

Comment:
I just realized, that after changing the STAT variable to ErrStat2, we might should replace this line with something like `CALL CheckError(ErrID_Fatal, ' Error allocating state vectors.')`
Otherwise, I think, the allocation error will be lost in the further course.

However, I'm unsure what's the best and clean git-like way to introduce this fix? Should I create a new feature branch just with this change and a PR from this afterwards?